v3/src/transports/http/handler.ts

```diff
@@ -0,0 +1,145 @@
+import { IncomingMessage, RequestListener, ServerResponse } from "http";
+import { IHttpRequest } from "./transport";
+import { processHttpRequest } from "./transport";
+import { GraphQLSchema } from "graphql";
+import { GraphQLRequest } from "../../types";
+
+/**
+ * A factory function that receives an instance of `ApolloServer` and returns
+ * a `RequestHandler` that can be used with Node.js' `http.createServer`, or
+ * Express' `app.use`.
+ *
+ * @param apollo An instance of `ApolloServer`
```

Comment:
Good catch, it is!  Though the previous implementation I had did leverage an instance of `ApolloServer`, I changed that to align with the expectations of the new `processGraphqlRequest` from #3428 

I'll adjust this, but speaking to the maintainability of these TSDoc notations though: This mismatch marks, in my experience on this new code-base, maybe the third case of TSDoc not aligning with the actual type signature.

While I do think using the TSDoc comment format to annotate functions and parameters makes a lot of sense (and displays really nicely in VSCode), I'm wondering if we need to use `@param` notation at all.

In particular, if I had placed this multi-line TSDoc comment above the type parameter in the actual function signature — and omitted the symbol name, it still seems to show up real nice in VSCode.  e.g.:

Rather than:

```ts
/**
 * Returns the average of two numbers.
 *
 * @param x - The first input number
 * @param y - The second input number
 * @returns The arithmetic mean of `x` and `y`
 *
 */
function getAverage(x, y) {
    return (x + y) / 2.0;
}

```

We might/could use:

```ts
/**
 * Returns the average of two numbers.
 *
 * @returns The arithmetic mean of `x` and `y`
 *
 */
function getAverage(
  /**
   * The first input number
   */
  x: number,
  /**
   * The second input number
   */
  y: number,
) {
    return (x + y) / 2.0;
}
```

Again, this seems to look nice in VSCode and shows as a nice-hover over on the parameter usages and Intelli-spidey-senses, however, TSDoc itself, seems to not compute it on its own.  That would be a sad if we ever generated documentation using that, but perhaps that's a feature that's bound to come to TSDoc eventually (or already has in some other form, but didn't seem to work on [their playground](https://microsoft.github.io/tsdoc/#).

Anyhow, just food for thought!  (On which I'm fishing for thoughts)

Reply:
Hmm… the second one maintains better, I agree, but I find the code harder to read. Between that and TSDoc using `@param`, I think I prefer it. But I also don't think it's something we can decide on a case-by-case basis, especially if there's a flag or something we can flip in TSDoc to have it pick up VSCode style comments.

In this case, we could have avoided the issue by not having `@param` in the first place, right? Given that "an instance of ApolloServer" tells you nothing that the types don't already.